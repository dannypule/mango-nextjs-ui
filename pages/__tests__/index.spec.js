// /* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import Main from '../index'

const initialProps = {
  scatterData: [
    {
      x: 30,
      y: 20
    },
    {
      x: 50,
      y: 70
    }
  ],
  getScatterData: () => {}
}

describe('Main', () => {
  describe('@render', () => {
    it('renders in default state', () => {
      const wrapper = shallow(<Main {...initialProps} />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
