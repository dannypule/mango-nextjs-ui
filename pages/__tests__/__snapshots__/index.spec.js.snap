// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main @render renders in default state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WrappedCmp
    getScatterData={[Function]}
    scatterData={
      Array [
        Object {
          "x": 30,
          "y": 20,
        },
        Object {
          "x": 50,
          "y": 70,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Connect(Component)
        getScatterData={[Function]}
        scatterData={
          Array [
            Object {
              "x": 30,
              "y": 20,
            },
            Object {
              "x": 50,
              "y": 70,
            },
          ]
        }
      />,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "getScatterData": [Function],
        "scatterData": Array [
          Object {
            "x": 30,
            "y": 20,
          },
          Object {
            "x": 50,
            "y": 70,
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Connect(Component)
          getScatterData={[Function]}
          scatterData={
            Array [
              Object {
                "x": 30,
                "y": 20,
              },
              Object {
                "x": 50,
                "y": 70,
              },
            ]
          }
        />,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "getScatterData": [Function],
          "scatterData": Array [
            Object {
              "x": 30,
              "y": 20,
            },
            Object {
              "x": 50,
              "y": 70,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
